pkg/codec/codec.go

```diff
@@ -8,5 +8,5 @@ import (
 	"github.com/pion/mediadevices/pkg/prop"
 )
 
-type VideoEncoderBuilder func(r video.Reader, p prop.Video) (io.ReadCloser, error)
-type AudioEncoderBuilder func(r audio.Reader, p prop.Audio) (io.ReadCloser, error)
+type VideoEncoderBuilder func(r video.Reader, p prop.Video, c CodecParameters) (io.ReadCloser, error)
+type AudioEncoderBuilder func(r audio.Reader, p prop.Audio, c CodecParameters) (io.ReadCloser, error)
```

Comment:
I think we should change the signature to `prop.Media` instead. This way, it's more consistent with what we already have in `driver`.

From:
```
type VideoEncoderBuilder func(r video.Reader, p prop.Video) (io.ReadCloser, error)
type AudioEncoderBuilder func(r audio.Reader, p prop.Audio) (io.ReadCloser, error)
```

To:
```
type VideoEncoderBuilder func(r video.Reader, p prop.Media) (io.ReadCloser, error)
type AudioEncoderBuilder func(r audio.Reader, p prop.Media) (io.ReadCloser, error)
```

And, update `prop.Media` to include `prop.Codec`:

```
type Media struct {
	Video
	Audio
	Codec
}

type Codec struct {
	// Target bitrate in bps.
	BitRate int

	// Quolity of the encoding [0-9].
	// Larger value results higher quality and higher CPU usage.
	// It depends on the selected codec.
	Quality int

	// Expected interval of the keyframes in frames.
	KeyFrameInterval int
}
```

What do you think?

Reply:
One of the concern to put codec parameters into there was that they are not constraints to select a device.
But it's fine as it makes the code simpler.